### PR TITLE
Response body writer example

### PR DESCRIPTION
**Pull request by adam-fowler**
Echo's request body and uses middleware to add SHA256 of the response body

### CODE REVIEW

**Review by Joannis — APPROVED** [review]
Nice and clean! I like it.

File: response-body-processing/Package.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import PackageDescription
+
+let package = Package(
+    name: "ResponseBodyProcessing",
+    platforms: [.macOS(.v14), .iOS(.v17), .tvOS(.v17)],
+    products: [
+        .executable(name: "App", targets: ["App"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/hummingbird-project/hummingbird.git", from: "2.0.0-rc.1"),
```
Comment (Joannis) [inline response-body-processing/Package.swift:13]: Can we bump this a little though?

**Comment by Joannis** [comment]
CI still failing

File: response-body-processing/Tests/AppTests/AppTests.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    func testApp() async throws {
+        let args = TestArguments()
+        let app = try await buildApplication(args)
+        let buffer = Self.randomBuffer(size: 256 * 000)
+        let digest = SHA256.hash(data: Data(buffer.readableBytesView))
+        try await app.test(.live) { client in
+            try await client.execute(uri: "/echo", method: .get, body: buffer) { response in
+                XCTAssertEqual(
+                    response.trailerHeaders?[.digest],
```
Comment (Joannis) [inline response-body-processing/Tests/AppTests/AppTests.swift:30]: Seems like we have a bug in the Hummingbird unit test client here. This should definitely yield a result if I see the code correctly

File: response-body-processing/Package.swift (hunk 1 above)
Comment (adam-fowler) [reply response-body-processing/Package.swift]: fixed

**Pull request merged**